### PR TITLE
[mongodb] bulkWrite operations types

### PR DESCRIPTION
**Pull request by avaly**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.mongodb.com/manual/reference/method/db.collection.bulkWrite/
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

FYI: I was forced to bump up the minimum TS version to 3.2 - it fails under 3.0 and 3.1.

### CODE REVIEW

File: types/mongodb/tsconfig.json [hunk 1]
```diff
@@ -23,13 +23,14 @@
...
+        "test/collection/distinct.ts",
         "test/collection/filterQuery.ts",
         "test/collection/findX.ts",
         "test/collection/insertX.ts",
         "test/collection/updateX.ts",
-        "test/collection/mapReduce.ts",
-        "test/collection/aggregate.ts",
-        "test/collection/count.ts",
-        "test/collection/distinct.ts"
+        "test/collection/mapReduce.ts"
```
Comment (LinusU) [inline types/mongodb/tsconfig.json:34]: Neat 👍

**Review by LinusU — APPROVED** [review]
This is amazing! Great work 👍

**Comment by avaly** [comment]
Does anyone have any idea how to solve the weird showing up in CI?

```
1> ERROR: 52:1    expect  TypeScript@3.9: Expected an error on this line, but found none.
1> ERROR: 56:21   expect  TypeScript@3.9 compile error: 
1> Type 'number' is not assignable to type 'string'.
```

It seems like TS v3.9 is reporting the error at a different line than the other versions, so the `$ExpectError` does not silence the error correctly under that TS version.

**Comment by avaly** [comment]
@LinusU got any clue how I can solve the CI issue in TS 3.9?

**Comment by peterblazejewicz** [comment]
> any clue how I can solve the CI issue in TS 3.9?

Try this, it will require some keyboard gimastics though, sample:
```diff
Last login: Sat Apr 18 17:53:52 on ttys000
^[^[nvm is not compatible with the npm config "prefix" option: currently set to "/usr/local/Cellar/node/13.5.0"
Run `nvm use --delete-prefix v12.13.1 --silent` to unset it.
Now using node v12.13.1 (npm v6.13.7)
➜  mongodb git:(pr/43930) git diff 
➜  mongodb git:(pr/43930) ✗ git diff
         },
     ]);
-    collectionType.bulkWrite([
-        // $ExpectError
-        {
-            updateMany: {
-                filter: { stringField: 'foo' },
-                update: {
-                    $set: { numberField: 'bar' },
+    // prettier-ignore
+    collectionType.bulkWrite([{ updateMany: { filter: { stringField: 'foo' }, update: { $set: { numberField: 'bar' }, // $ExpectError
                 },
             },
         },
...
-    collectionType.bulkWrite([
-        // $ExpectError
-        {
-            updateMany: {
-                filter: { stringField: 123 },
+    // prettier-ignore
+    collectionType.bulkWrite([{ updateMany: { filter: { stringField: 123 }, // 
$ExpectError
                 update: {
                     $set: { numberField: 123 },
                 },
             },
         },
     ]);
```

**Review by LinusU — APPROVED** [review]
👍

**Comment by elibarzilay** [comment]
@avaly, there are merge conflicts...?

**Review by LinusU — APPROVED** [review]

**Pull request merged**